client/state/selectors/get-current-plan-purchase-id.js

```diff
@@ -0,0 +1,33 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { get, isNaN } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getCurrentPlan } from 'state/sites/plans/selectors';
+
+/**
+ * Returns a site's current plan purchase ID or null if the site doesn't exist or the purchase ID
+ * is unknown.
+ *
+ * The ID returned by this selector is retrieved from the site plan. It is intended to provide a
+ * means of retrieving full purhcase information based on a site and it's plan information.
+ * Caution! It _does not_ retrieve the ID from a purchase.
```

Comment:
Well, it's still retrieving the ID of a purchase, and whether it's doing it from the purchases tree or the sites tree - it's still the ID of a purchase, right? This is what makes the comment confusing to me.

Personally I'd remove the comment, because selectors are usually simple enough and it's easy to see which tree data is pulled from. And I'm pretty sure there are other cases we have data that can be pulled from separate trees (like user capabilities, or Jetpack module data for example), but we don't have such clarifications. 

If you still insist on having a comment there, maybe it would be more useful to mention the state tree that we use, and the state tree that we're not using - that would probably provide more information to anyone, willing to use this selector.